lib/search/hotel.js

```diff
@@ -127,47 +127,41 @@ function computeMarketingQuery (fastQuery, marketing) {
 /**
  * Executes each query group. (Query groups are based on geo.)
  */
-function query (queries, hotelsToIntersect, callback) {
+function query (queries, hotelsToAdd, callback) {
   async.parallel(queries.map(function (query) {
     return computeQueryFunctions(query, database);
   }), function (err, result) {
     err = err || null; // if no error return null
+    // Intersect the search results together
+    var intersected = intersectResults(result);
+    if (hotelsToAdd.length) intersected.data.push(filterUniqueHotels(hotelsToAdd, intersected.hotelIds));
+    return callback(err, {queries: intersected.queries, data: intersected.data});
+  });
+}
 
-    var returnObject = {
-      queries: [],
-      data: []
-    };
+function filterUniqueHotels (hotels, hotelIdsToFilter) {
+  return hotels.map(function (hotel) {
+    if (~hotelIdsToFilter.indexOf(hotel.properties.id)) return hotel;
```

Comment:
not everyone will *understand* the `~` ...
http://stackoverflow.com/questions/12299665/what-does-a-tilde-do-when-it-precedes-an-expression
any reason not to use `!` ?